Content.Server/AbstractAnalyzer/AbstractAnalyzerComponent.cs

```diff
@@ -0,0 +1,68 @@
+using Robust.Shared.Audio;
+
+namespace Content.Server.AbstractAnalyzer;
+
+/// <summary>
+/// After scanning, retrieves the target Uid to use with its related UI.
+/// </summary>
+/// <remarks>
+/// Requires <c>ItemToggleComponent</c>.
+/// </remarks>
+public abstract partial class AbstractAnalyzerComponent : Component
+{
+    /// <summary>
+    /// When should the next update be sent for the patient
+    /// </summary>
+    /// <remarks>
+    /// Override this in your implementation with:
+    ///
+    /// ```cs
+    /// [DataField(customTypeSerializer: typeof(TimeOffsetSerializer))]
+    /// [AutoPausedField]
+    /// public override TimeSpan NextUpdate { get; set; } = TimeSpan.Zero;
+    /// ```
+    /// </remarks>
+    public abstract TimeSpan NextUpdate { get; set; }
```

Comment:
why is this `abstract`? it should just be a regular datafield manipulated by the relevant systems instead of having any logic in the components

Reply:
This is abstract because the children need `[AutoPausedField]` which does not inherit correctly.